drivers/pinctrl/Kconfig.npcx

```diff
@@ -0,0 +1,15 @@
+# NPCX Pin Controller configuration options
+
+# Copyright (c) 2022 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_ST_PINCTRL_NPCX := nuvoton,npcx-pinctrl
+
+config PINCTRL_NPCX
+	bool "Nuvoton NPCX embedded controller (EC) pin controller driver"
+	depends on SOC_FAMILY_NPCX
+	select PINCTRL_STORE_REG
```

Comment:
Non-blocking comment - this adds 4 bytes to every pinctrl node correct?  But this is only needed by the PWM nodes.

If you can repack the bitfields of `npcx_periph` to use 3 bits to store the PWM instance number.  You could then store the PWM base addresses into `npcx_pinctrl_cfg` similar to the `base_scfg` field.

Using PINCTRL_STORE_REG is a cleaner approach, so you would need to evaluate how much flash space is saved by encoding the PWM instance in place of using `reg`.

Reply:
> Non-blocking comment - this adds 4 bytes to every pinctrl node correct? But this is only needed by the PWM nodes.

ML> If you mean each given device that applies pinctrl mechanism will increase 4 bytes, yes, you're correct.


> If you can repack the bitfields of `npcx_periph` to use 3 bits to store the PWM instance number. You could then store the PWM base addresses into `npcx_pinctrl_cfg` similar to the `base_scfg` field.
> Using PINCTRL_STORE_REG is a cleaner approach, so you would need to evaluate how much flash space is saved by encoding the PWM instance in place of using `reg`.

ML> I have thought about that before. But if we have a chance to add drive mode support for the other peripheral devices, ```PINCTRL_STORE_REG``` is a clearer way to maintain pinctrl mechanism. As I mentioned above, the flash space increases by 4 bytes for every device using ```pinctl-names``` and ```proctrl-0/1/2``` properties for pinctrl mechanism. If we need to save flash spaces in the furture, I prefer to create the other PR that includes:
1. Restore device's label name in ```npcx_periph```
2. Use ```device_get_binding``` to get the correct register address.

But the cost is the latency of selecting drive mode is longer.

Reply:
Note that `device_get_binding` is rarely needed these days, `DEVICE_DT_GET` can be used for most situations. It is sometimes not necessary to store a device pointer either, see e.g. how GD32 pinctrl driver handles selected port.